### DISCUSSION TITLE
Screen Img streeming to rtsp

### DISCUSSION THREAD

**Discussion by DenisGimranov** [ngraziano/SharpRTSP#88] [Q&A]
Tell me how to substitute your picture in RTSPCameraExample instead of a gray background with date and time, I tried to replace yuv_data by substituting

 ```
Screen screen = Screen.PrimaryScreen;
Bitmap bitmap = new Bitmap(screen.Bounds.Width, screen.Bounds.Height);
Graphics graphics = Graphics.FromImage(bitmap);
graphics.CopyFromScreen(screen.Bounds.X, screen.Bounds.Y, 0, 0, screen.Bounds.Size);
Image image = (Image)bitmap;
System.IO.MemoryStream memoryStream = new System.IO.MemoryStream();
image.Save(memoryStream, System.Drawing.Imaging.ImageFormat.Bmp);

byte[] raw_video_nal = h264_encoder.CompressFrame(memoryStream.ToArray());
```

I understand that this is not the right way, but I can’t understand how yuv_data appears at all during the test demo call

```
TestCard av_source = new TestCard((int)width, (int)height, (int)fps);
av_source.ReceivedYUVFrame += Video_source_ReceivedYUVFrame; 
```
As I understand it, I need to make changes to this place and make my own screen, but how?

**Comment by RogerHardiman** [comment]
There is a worker thread that makes YUV images and uses a C# event handler (the += myhandler type code).
In the handler it compresses the image and sends it.
But note that a Bitmap is often RGB and not YUV. You'll need to convert it into YUV format first

**Reply by DenisGimranov** [reply]
Would like to know how to do it)

Tell?

**Comment by RogerHardiman** [comment]
Quick suggestions.
1) have a C# Timer that fires an event every 100ms.
2) in that timer handler you can take you RGB bitmap and you need to convert it to a YUV array in YUV420 format. Google will find examples of the maths needed to convert RGB into YUV420
3) then you can call the compress function with your YUV array

Don't expect it to be really fast. On commercial projects I do all the hard work calling optimised C code.

**Reply by DenisGimranov** [reply]
Ready to discuss https://t.me/GimDenis a commercial option, because still weak in C#, but the solution is necessary for my small business

**Comment by RogerHardiman** [comment]
For me, I have no spare time for any extra commercial options. (already working full time on other projects. Sorry)
Perhaps someone else will be able to help

**Reply by DenisGimranov** [reply]
I really hope so ;)
